--- _deps/cmakebuild-src/cmake/build_external/BuildLibrett.cmake.org	2025-02-08 20:48:05.784749660 -0800
+++ _deps/cmakebuild-src/cmake/build_external/BuildLibrett.cmake	2025-02-08 20:55:17.847113982 -0800
@@ -23,8 +23,8 @@
 )
 else()
 ExternalProject_Add(Librett_External
-    GIT_REPOSITORY https://github.com/victor-anisimov/Librett.git
-    GIT_TAG ${LIBRETT_GIT_TAG}
+    GIT_REPOSITORY https://github.com/edoapra/Librett.git
+    GIT_TAG 0eb41dc689e30019118b5a484171b7fcd0ebf1b7
     UPDATE_DISCONNECTED 1
     CMAKE_ARGS ${DEPENDENCY_CMAKE_OPTIONS} -DENABLE_SYCL=${${CMSB_PROJECTS}_HAS_DPCPP}
     -DENABLE_CUDA=${${CMSB_PROJECTS}_HAS_CUDA} -DENABLE_HIP=${${CMSB_PROJECTS}_HAS_HIP} 
